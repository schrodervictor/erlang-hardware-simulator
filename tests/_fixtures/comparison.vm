push constant 10
push constant 10
eq
push constant 20
push constant 30
lt
push constant 30
push constant 20
gt
