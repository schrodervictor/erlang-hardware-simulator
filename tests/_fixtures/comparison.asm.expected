// push constant 10
@10
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 10
@10
D=A
@SP
A=M
M=D
@SP
M=M+1
// eq
@SP
AM=M-1
D=M
A=A-1
D=M-D
@comparison.2.PUSH.TRUE
D;JEQ
// (comparison.2.PUSH.FALSE)
    D=0
    @comparison.2.PUSH.END
    0;JMP
(comparison.2.PUSH.TRUE)
    D=-1
(comparison.2.PUSH.END)
    @SP
    A=M-1
    M=D
// push constant 20
@20
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 30
@30
D=A
@SP
A=M
M=D
@SP
M=M+1
// lt
@SP
AM=M-1
D=M
A=A-1
D=M-D
@comparison.5.PUSH.TRUE
D;JLT
// (comparison.5.PUSH.FALSE)
    D=0
    @comparison.5.PUSH.END
    0;JMP
(comparison.5.PUSH.TRUE)
    D=-1
(comparison.5.PUSH.END)
    @SP
    A=M-1
    M=D
// push constant 30
@30
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 20
@20
D=A
@SP
A=M
M=D
@SP
M=M+1
// gt
@SP
AM=M-1
D=M
A=A-1
D=M-D
@comparison.8.PUSH.TRUE
D;JGT
// (comparison.8.PUSH.FALSE)
    D=0
    @comparison.8.PUSH.END
    0;JMP
(comparison.8.PUSH.TRUE)
    D=-1
(comparison.8.PUSH.END)
    @SP
    A=M-1
    M=D
